dynamometer-workload/src/main/java/com/linkedin/dynamometer/workloadgenerator/AbusiveUserMapper.java

```diff
@@ -0,0 +1,159 @@
+  private int mappersPerUser;
+  private boolean enableWriteOps;
+
+  @Override
+  public String getDescription() {
+    return "This mapper creates 1-byte files for the specified duration.";
+  }
+
+  @Override
+  public List<String> getConfigDescriptions() {
+    List<String> baseList = TimedInputFormat.getConfigDescriptions();
+    baseList.add(FILE_PARENT_PATH_KEY + " (default: " + FILE_PARENT_PATH_DEFAULT +
+        "): The root directory for the job to create files in.");
+    baseList.add(NUM_LARGE_JOBS_KEY + " (default: " + NUM_LARGE_JOBS_DEFAULT +
+        "): Number of large listing jobs.");
+    baseList.add(NUM_FILES_KEY + " (default: " + NUM_FILES_DEFAULT +
+        "): Number of files in the directory for the large listing operations.");
+    baseList.add(MAPPERS_PER_USER_KEY + " (default: " + MAPPERS_PER_USER_DEFAULT +
+        "): Number of mappers per user.");
+    baseList.add(ENABLE_WRITE_OPS_KEY + " (default: " + ENABLE_WRITE_OPS_DEFAULT +
+        "): For the small operation, creates a single directory instead of listing a single file directory.");
+    return baseList;
+  }
+
+  @Override
+  public boolean verifyConfigurations(Configuration conf) {
+    return TimedInputFormat.verifyConfigurations(conf);
+  }
+
+  @Override
+  public void setup(Context context) throws IOException {
+    // Load config
+    Configuration conf = context.getConfiguration();
+    int taskID = context.getTaskAttemptID().getTaskID().getId();
+    numLargeJobs = conf.getInt(NUM_LARGE_JOBS_KEY, NUM_LARGE_JOBS_DEFAULT);
+    int numFiles = conf.getInt(NUM_FILES_KEY, NUM_FILES_DEFAULT);
+    mappersPerUser = conf.getInt(MAPPERS_PER_USER_KEY, MAPPERS_PER_USER_DEFAULT);
+    enableWriteOps = conf.getBoolean(ENABLE_WRITE_OPS_KEY, ENABLE_WRITE_OPS_DEFAULT);
+
+    // Load filesystem
+    String namenodeURI = conf.get(WorkloadDriver.NN_URI);
+    UserGroupInformation loginUser = UserGroupInformation.getLoginUser();
+    String proxyUser = "fakeuser" + taskID / mappersPerUser;
+    UserGroupInformation ugi = UserGroupInformation.createProxyUser(proxyUser, loginUser);
+    FileSystem fs = ugi.doAs(new PrivilegedAction<FileSystem>() {
+      @Override
+      public FileSystem run() {
+        try {
+          FileSystem fs = new DistributedFileSystem();
+          fs.initialize(URI.create(namenodeURI), conf);
+          return fs;
+        } catch (IOException ioe) {
+          throw new RuntimeException(ioe);
+        }
+      }
+    });
+    fsMap.put(proxyUser, fs);
+
+    // Load default path
+    String fileParentPath = conf.get(FILE_PARENT_PATH_KEY, FILE_PARENT_PATH_DEFAULT);
+    parentFolder = new Path(new Path(fileParentPath), "mapper" + taskID);
+
+    // Make job folder
+    fs.mkdirs(parentFolder);
+
+    if (taskID < numLargeJobs) {
+      // Large job: set up files for large listing op
+      for (int i = 0; i < numFiles; i++) {
+        fs.mkdirs(new Path(parentFolder, new Path("big" + i)));
+      }
+    } else if (enableWriteOps) {
```

Comment:
Probably not necessary but clarifies the structure a bit and follows the pattern of the `map` step.